src/main/java/org/broadinstitute/hellbender/tools/exome/AllelicCount.java

```diff
@@ -49,9 +79,61 @@ public AllelicCount(final SimpleInterval interval, final int refReadCount, final
 
     public int getAltReadCount() { return altReadCount; }
 
-    public Nucleotide getRefNucleotide() { return refNucleotide; }
+    public Nucleotide getRefNucleotide() {
+        return Utils.nonNull(refNucleotide, "The Ref nucleotide information is not available.");
+    }
+
+    public Nucleotide getAltNucleotide() {
+        return Utils.nonNull(altNucleotide, "The Alt nucleotide information is not available.");
+    }
+
+    public Integer getReadDepth() {
+        return Utils.nonNull(readDepth, "The read depth information is not available.");
+    }
+
+    public Double getHetLogOdds() {
+        return Utils.nonNull(hetLogOdds, "The log odds of heterozygosity is not available.");
+    }
 
-    public Nucleotide getAltNucleotide() { return altNucleotide; }
+    public static final Function<DataLine, AllelicCount> basicParser = dataLine -> {
```

Comment:
Also you may need to move these methods / lambdas to the new ```AllelicCountReader``` if you follow the review below.